src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -56,81 +56,108 @@ class SessionService(
-    }
-
+    val noAssociationConflictSessions = getNoAssociationConflictSessions(sessions, prisonerId)
+    sessions = filterPrisonerConflict(sessions, prisonerId, noAssociationConflictSessions)
+    populateConflict(sessions, prisonerId, noAssociationConflictSessions)
     populateBookedCount(sessions)
 
     return sessions.sortedWith(compareBy { it.startTimestamp })
   }
 
-  fun filterByCategory(sessionTemplates: List<SessionTemplate>, prisonerCategory: String?): List<SessionTemplate> {
-    val hasSessionsWithCategoryGroups = sessionTemplates.any { it.permittedSessionCategoryGroups.isNotEmpty() }
-    if (hasSessionsWithCategoryGroups) {
-      return sessionTemplates.filter { sessionTemplate ->
-        sessionValidator.isSessionAvailableToPrisonerCategory(prisonerCategory, sessionTemplate)
-      }
-    }
+  private fun getDateRange(
+    prison: Prison,
+    minOverride: Int? = null,
+    maxOverride: Int? = null,
+  ): DateRange {
+    val today = LocalDate.now()
+
+    val min = minOverride ?: prison.policyNoticeDaysMin
+    val max = maxOverride ?: prison.policyNoticeDaysMax
 
-    return sessionTemplates
+    val requestedBookableStartDate = today.plusDays(min.toLong())
+    val requestedBookableEndDate = today.plusDays(max.toLong())
+    return DateRange(requestedBookableStartDate, requestedBookableEndDate)
+  }
+
+  private fun filterByCategory(
+    sessionTemplate: SessionTemplate,
+    sessionTemplates: List<SessionTemplate>,
+    prisonerCategory: String?,
+  ): Boolean {
+    val hasSessionsWithCategoryGroups = sessionTemplates.any { sessionTemplate.permittedSessionCategoryGroups.isNotEmpty() }
+    return if (hasSessionsWithCategoryGroups) {
+      sessionValidator.isSessionAvailableToPrisonerCategory(prisonerCategory, sessionTemplate)
+    } else {
+      true
+    }
   }
 
-  fun filterByIncentiveLevels(
+  private fun filterByIncentiveLevels(
+    sessionTemplate: SessionTemplate,
     sessionTemplates: List<SessionTemplate>,
     prisonerIncentiveLevel: IncentiveLevel?,
-  ): List<SessionTemplate> {
+  ): Boolean {
     val hasSessionsWithIncentiveLevelGroups =
       sessionTemplates.any { it.permittedSessionIncentiveLevelGroups.isNotEmpty() }
-    if (hasSessionsWithIncentiveLevelGroups) {
-      return sessionTemplates.filter { sessionTemplate ->
-        sessionValidator.isSessionAvailableToIncentiveLevel(prisonerIncentiveLevel, sessionTemplate)
-      }
+    return if (hasSessionsWithIncentiveLevelGroups) {
+      sessionValidator.isSessionAvailableToIncentiveLevel(prisonerIncentiveLevel, sessionTemplate)
+    } else {
+      true
     }
+  }
 
-    return sessionTemplates
+  private fun getAllSessionTemplatesForDateRange(prisonCode: String, dateRange: DateRange): List<SessionTemplate> {
+    return sessionTemplateRepository.findSessionTemplateMinCapacityBy(
+      prisonCode = prisonCode,
+      rangeStartDate = dateRange.fromDate,
+      rangeEndDate = dateRange.toDate,
+    )
+  }
+
+  fun filterByLocation(
```

Comment:
Is this  filter now or should it be name differently as it returns a Boolean

Reply:
Updated method names

Reply:
Nice isAvailableToLocation